From d8e14d58fddd85a84e58114163f3449c503ff7e2 Mon Sep 17 00:00:00 2001
From: yuric <yuric@marvell.com>
Date: Mon, 8 Apr 2019 15:23:11 +0300
Subject: [PATCH 146/386] net: mvpp2: remove destination queue hard coded map

Remove hard coded destination queue map, destination queue should be
defined by Classifier rules.
The patch fixes MUSDK Classifier functionality.

Change-Id: I2eb78d6d551c348889bd94cd4b880dd2743536fd
Signed-off-by: yuric <yuric@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7094
Reviewed-by: Liron Himi <liron.himi@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
index 16018ed94b3e..226a9f433ae6 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
@@ -915,6 +915,11 @@ void mvpp2_cls_init(struct mvpp2 *priv)
 		mvpp2_cls_lookup_write(priv, &le);
 	}
 
+	/* Clear CLS_SWFWD_PCTRL register - value of QueueHigh is defined by
+	 * the Classifier
+	 */
+	mvpp2_write(priv, MVPP2_CLS_SWFWD_PCTRL_REG, 0);
+
 	mvpp2_cls_port_init_flows(priv);
 
 	/* Initialize C2 */
@@ -993,17 +998,8 @@ void mvpp22_rss_disable(struct mvpp2_port *port)
 /* Set CPU queue number for oversize packets */
 void mvpp2_cls_oversize_rxq_set(struct mvpp2_port *port)
 {
-	u32 val;
-
 	mvpp2_write(port->priv, MVPP2_CLS_OVERSIZE_RXQ_LOW_REG(port->id),
 		    port->first_rxq & MVPP2_CLS_OVERSIZE_RXQ_LOW_MASK);
-
-	mvpp2_write(port->priv, MVPP2_CLS_SWFWD_P2HQ_REG(port->id),
-		    (port->first_rxq >> MVPP2_CLS_OVERSIZE_RXQ_LOW_BITS));
-
-	val = mvpp2_read(port->priv, MVPP2_CLS_SWFWD_PCTRL_REG);
-	val |= MVPP2_CLS_SWFWD_PCTRL_MASK(port->id);
-	mvpp2_write(port->priv, MVPP2_CLS_SWFWD_PCTRL_REG, val);
 }
 
 static inline u32 mvpp22_rxfh_indir(struct mvpp2_port *port, u32 rxq)
-- 
2.17.1

